File: sdk/src/test/java/com/cxense/cxensesdk/model/ExternalItemTest.java
package com.cxense.cxensesdk.model;

import com.cxense.cxensesdk.BaseTest;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2017-07-31).
 */
public class ExternalItemTest extends BaseTest {
    private ExternalItem externalItem;

    @Before
    public void setUp() throws Exception {
        super.setUp();
        externalItem = new ExternalItem("group", "item");
    }

    @Test
    public void getGroup() throws Exception {
        String group = "group";
        externalItem.group = group;
        assertEquals(group, externalItem.getGroup());
    }

    @Test
    public void getItem() throws Exception {
        String item = "item";
        externalItem.group = item;
        assertEquals(item, externalItem.getItem());
    }
}

File: sdk/src/test/java/com/cxense/cxensesdk/model/ExternalItemTest.kt
package com.cxense.cxensesdk.model

import com.cxense.cxensesdk.assertFailsWithMessage
import kotlin.test.Test

class ExternalItemTest {
    @Test
    fun withInvalidGroup() {
        assertFailsWithMessage<IllegalArgumentException>("Group should not be empty", "Expected fail for group") {
            ExternalItem("", "")
        }
    }

    @Test
    fun withInvalidItem() {
        assertFailsWithMessage<IllegalArgumentException>("Item can't be longer", "Expected fail for item") {
            ExternalItem("group", "q".repeat(500))
        }
    }

    @Test
    fun createValid() {
        ExternalItem("group", "item")
    }
}
